### PR TITLE
Task 33321557: Streamline XSAPI thunk dll generation and ship it in the GDK

### PR DESCRIPTION
**Pull request by tculotta**

### CODE REVIEW

**Review by rakhad — APPROVED** [review]

File: Include/httpClient/config.h [hunk 1]
```diff
@@ -35,6 +35,15 @@
         #define HC_PLATFORM HC_PLATFORM_GDK
     #elif !defined(WINAPI_FAMILY) || (WINAPI_FAMILY == WINAPI_FAMILY_DESKTOP_APP)
         #define HC_PLATFORM HC_PLATFORM_WIN32
+        #include <windef.h>
```
Comment (jasonsandlin) [inline Include/httpClient/config.h:38]: isn't this the XAL win32 section?  seems odd to need to put it here

**Comment by tculotta** [comment]
Also changed to check against SDK version (this code was introduced in vibranium), so the #include is no longer needed which was problematic anyway.

File: Include/httpClient/config.h (hunk 1 above)
Comment (tculotta) [reply Include/httpClient/config.h]: moved to correct block

**Review by jasonsandlin — APPROVED** [review]
:shipit:

**Pull request merged**